src/Containers/CheckoutPage/PaymentPage/PaymentPage.js

```diff
@@ -30,60 +30,89 @@ const PaymentPage=()=>{
-                                {paymentMode==="CREDITCARD"&&<CardDetails
-                                    inputRef={inputRef}
-                                    nameOnCard={nameOnCard}
-                                    setNameOnCard={event=>setNameOnCard(event.target.value)}
-                                    cardNumber={cardNumber}
-                                    setCardNumber={event=>setCardNumber(event.target.value)}
-                                    expirationDate={expirationDate}
-                                    setExpirationDate={event=>setExpirationDate(event.target.value)}
-                                    cvv={cvv}
-                                    setCvv={event=>setCvv(event.target.value)}
-                                />}
-                            <FormControlLabel value="DEBITCARD" control={<Radio color="primary"/>} label="Debit Card"/>
-                                {paymentMode==="DEBITCARD"&&<CardDetails
-                                    inputRef={inputRef}
-                                    nameOnCard={nameOnCard}
-                                    setNameOnCard={event=>setNameOnCard(event.target.value)}
-                                    cardNumber={cardNumber}
-                                    setCardNumber={event=>setCardNumber(event.target.value)}
-                                    expirationDate={expirationDate}
-                                    setExpirationDate={event=>setExpirationDate(event.target.value)}
-                                    cvv={cvv}
-                                    setCvv={event=>setCvv(event.target.value)}
-                                />}
-                            <FormControlLabel value="COD" control={<Radio color="primary"/>} label="Cash on Delivery"/>
-                        </RadioGroup>
-                    </FormControl>
-                </div>
-                <button
-                    className={`${classes["button-solid"]} ${classes["button-primary"]}`}
-                    type="submit"
-                >
-                    Continue
-                </button>
-            </form>
+            <div className={classes["form-container"]}>
+                <FormControl component="fieldset">
+                    <RadioGroup aria-label="Payment mode" name="payment mode" value={paymentMode} onChange={event=>{
+                            setPaymentMode(event.target.value)
+                            if(event.target.value==="COD")
+                                dispatch({
+                                    type:"ADD_PAYMENT_DETAILS",
+                                    payload:"COD"
+                                })
+                            else
+                                dispatch({
+                                    type:"ADD_PAYMENT_DETAILS",
+                                    payload:null
+                                })
+                        }}>
```

Comment:
i think we should extract the onChange function outside, so that the jsx looks cleaner